UOP1_Project/Assets/Scripts/Quests/Editor/QuestEditorWindow.cs

```diff
@@ -0,0 +1,289 @@
+    public static void ShowWindow()
+    {
+        QuestEditorWindow wnd = GetWindow<QuestEditorWindow>();
+        wnd.titleContent = new GUIContent("QuestEditorWindow");
+    }
+
+    public void CreateGUI()
+    {
+        VisualElement root = rootVisualElement;
+
+        // Import UXML
+        var visualTree = AssetDatabase.LoadAssetAtPath<VisualTreeAsset>("Assets/Scripts/Quests/Editor/QuestEditorWindow.uxml");
+        root.Add(visualTree.CloneTree());
+
+        //Add Image
+        VisualElement preview = root.Q<VisualElement>("actor-preview");
+        actorPreview = new Image();
+        preview.Add(actorPreview);
+
+        //Import USS
+        var styleSheet = AssetDatabase.LoadAssetAtPath<StyleSheet>("Assets/Scripts/Quests/Editor/QuestEditorWindow.uss");
+        root.styleSheets.Add(styleSheet);
+
+        //Register button event
+        Button refreshQuestPreviewBtn = root.Q<Button>("refresh-preview-btn");
+        refreshQuestPreviewBtn.RegisterCallback<ClickEvent>((evt) => SetUpQuestPreview(currentSeletedQuest));
+
+        LoadAllQuestsData();
+    }
+
+    private void LoadAllQuestsData()
+    {
+	    //Load all questlines
+	    FindAllSOByType(out QuestlineSO[] questLineSOs);
+	    RefreshListView(out ListView allQuestlinesListView, "questlines-list", questLineSOs);
+
+	    allQuestlinesListView.onSelectionChanged += (questlineEnumerable) =>
+	    {
+		    QuestlineSO questLine = GetDataFromListViewItem<QuestlineSO>(questlineEnumerable);
+		    RefreshListView(out ListView allQuestsListView, "quests-list", questLine.Quests.ToArray());
+
+		    allQuestsListView.onSelectionChanged += (questEnumerable) =>
+		    {
+			    currentSeletedQuest = GetDataFromListViewItem<QuestSO>(questEnumerable);
+			    RefreshListView(out ListView allStepsListView, "steps-list",currentSeletedQuest.Steps.ToArray() );
+
+			    SetUpQuestPreview(currentSeletedQuest);
+
+			    allStepsListView.onSelectionChanged += (stepEnumerable) =>
+			    {
+				    StepSO step = GetDataFromListViewItem<StepSO>(stepEnumerable);
+				    DisplayAllProperties(step, "step-info-scroll");
+
+				    //Find all DialogueDataSOs in the same folder of the StepSO
+				    FindAllSOsInTargetFolder(step, out DialogueDataSO[] dialogueDataSOs);
+				    RefreshListView(out ListView dialoguesListView, "dialogues-list", dialogueDataSOs);
+
+				    dialoguesListView.onSelectionChanged += (dialogueEnumerable) =>
+				    {
+					    DialogueDataSO dialogueData = GetDataFromListViewItem<DialogueDataSO>(dialogueEnumerable);
+					    DisplayAllProperties(dialogueData, "dialogue-info-scroll");
+				    };
+			    };
+		    };
+	    };
+    }
+
+    private T GetDataFromListViewItem<T>(List<object> enumberable) where T : class
+    {
+	    T data = default;
+	    foreach (var item in enumberable)
+	    {
+		    data = item as T;
+	    }
+	    return data;
+    }
+
+    private void SetUpQuestPreview(QuestSO quest)
+    {
+	    if (quest != null)

```

Comment:
Again, you can reduce nesting here by doing something like:

`if (quest == null) return;
// rest of code`